### ISSUE TITLE
Filter gitignore'd files

### ISSUE THREAD

**Issue by borekb** [igorshubovych/markdownlint-cli#46]
As initially discussed in https://github.com/igorshubovych/markdownlint-cli/issues/44, it would be great if markdownlint could pic up ignore rules from `.gitignore` files. With a flag suggested by @DavidAnson, it could look like this:

```
markdownlint --gitignore
```

(I'd prefer to put it in a config file but that is a generic request, see #45.)

**Comment by MaxMilton** [comment]
I quite like ESLint's `--ignore-path` flag, e,g,:

```sh
eslint --ignore-path .gitignore .
```

It also follows linter conventions of supporting a `.eslintignore` file. Would be cool to support a `.markdownlintignore` file too since it would likely use the same logic as parsing and loading from `.gitignore`.

**Comment by borekb** [comment]
I now also prefer what @MaxMilton said, for example, for ESLint, I can do:

```
eslint --ignore-path .gitignore --ignore-path .prettierignore ...
```

which is actually what I'd use for markdownlint as well.

**Comment by DavidAnson** [comment]
With the commit for #61, `.markdownlintignore` is now supported as with `.eslintignore`. Support for `--ignore-path` could use this to allow people to point at different files, but I'm hopeful that `.markdownlintignore` is the right choice most of the time.
